src/ports/postgres/modules/svm/svm.py_in

```diff
@@ -66,211 +253,118 @@ def svm(schema_madlib, source_table, model_table,
+    else:
+        params_dict['decay_factor'] = params_dict['decay_factor'][0]
+
+    if not cv_params and params_dict['n_folds'] <= 1:
+        return
+
+    if cv_params and params_dict['n_folds'] <= 1:
+        plpy.error("SVM Error: parameters must be a scalar "
+                   "or of length 1 when n_folds is 0 or 1")
+        return
+
+    if not cv_params and params_dict['n_folds'] > 1:
+        plpy.warning('SVM Warning: no cross validate params provided! '
+                     'Ignore {}-folds cross validation request.'
+                     .format(params_dict['n_folds']))
+        return
+
+    scorer = 'classification' if args['is_svc'] else 'regression'
+    sub_args = {'params_dict':cv_params}
+    cv = CrossValidator(_svm_parsed_params,svm_predict,scorer,args)
+    val_res = cv.validate(sub_args, params_dict['n_folds']).sorted()
+    val_res.output_tbl(params_dict['validation_result'])
+    params_dict.update(val_res.first('sub_args')['params_dict'])
+
+
+def _svm_parsed_params(schema_madlib, source_table, model_table,
+                       dependent_varname, independent_varname, kernel_func,
+                       kernel_params, grouping_col, params_dict, is_svc,
+                       verbose, **kwargs):
+    """
+    Executes the linear support vector classification algorithm.
+    """
+    grouping_str = _verify_grouping(schema_madlib,
+                                    source_table,
+                                    grouping_col)
+
+    kernel_func = _verify_kernel(kernel_func)
+
+    # arguments for iterating
+    n_features = num_features(source_table,
+                              independent_varname)
+
+    args = {
+            'rel_args': unique_string(desp='rel_args'),
+            'rel_state': unique_string(desp='rel_state'),
+            'col_grp_iteration': unique_string(desp='col_grp_iteration'),
+            'col_grp_state': unique_string(desp='col_grp_state'),
+            'col_grp_key': unique_string(desp='col_grp_key'),
+            'col_n_tuples': unique_string(desp='col_n_tuples'),
+            'state_type': "double precision[]",
+            'n_features': n_features,
+            'verbose': verbose,
+            'schema_madlib': schema_madlib,
+            'grouping_str': grouping_str,
+            'grouping_col': grouping_col,
+            'rel_source': source_table,
+            'col_ind_var': independent_varname,
+            'col_dep_var': dependent_varname}
+
+    args.update(_verify_params_dict(params_dict))
+    args.update(_process_epsilon(is_svc, args))
+    args.update(_svc_or_svr(is_svc, source_table, dependent_varname))
+
+    # place holder for compatibility
+    plpy.execute("CREATE TABLE pg_temp.{0} AS SELECT 1".format(args['rel_args']))
+    # actual iterative algorithm computation
+    n_iters_run = _compute_svm(args)
+    _summary(n_iters_run, model_table, args)
+>>>>>>> b105d1c... SVM: Add cross validation support and generic CrossValidator class
```

Comment:
@mktal Invalid line. Also run the py files through a linter. There are a few places where we're not following the pep8 guidelines.